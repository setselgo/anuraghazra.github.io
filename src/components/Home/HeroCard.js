import React from 'react';
import { withTheme } from 'styled-components';

import {
  HeroCardWrapper,
  CodeCardWrapper,
  ColorPaletteWrapper,
  ColorBox,
} from './HeroCard.style'

export const CodeCard = () => {
  return (
    <CodeCardWrapper>
      <pre>1   class <b>Person</b> {'{'}</pre>
      <pre>2         constructor() {'{'}</pre>
      <pre>3             this.name = "<b>Anurag Hazra</b>";</pre>
      <pre>4             this.traits = ["<b>DESIGN</b>", "<b>DEV</b>"]</pre>
      <pre>5             this.age = new Date().getFullYear() - 2001;</pre>
      <pre>6         {"}"}</pre>
      <pre>7   {"}"}</pre>
    </CodeCardWrapper>
  )
}

export const ColorPalette = withTheme(({ theme }) => {
  return (
    <ColorPaletteWrapper>
      <ColorBox color={theme.primaryColor} />
      <ColorBox color={theme.secondaryColor} />
      <ColorBox color={theme.gradient} />
      <ColorBox color={theme.primaryBlack} />
      <ColorBox color={theme.accentColor} />
    </ColorPaletteWrapper>
  )
})

export const HeroCard = () => {
  return (
    <HeroCardWrapper>
      <CodeCard />
      <ColorPalette />
    </HeroCardWrapper>
  )
}