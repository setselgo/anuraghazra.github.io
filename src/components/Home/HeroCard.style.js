import styled from 'styled-components';

export const HeroCardWrapper = styled.div`
  position: relative;
`

export const CodeCardWrapper = styled.div`
  padding: 30px 35px;
  border-radius: 10px;
  background-color: ${props => props.theme.secondaryColor};
  box-shadow: ${props => props.theme.shadowSmall}; 

  pre {
    /* white-space: pre-wrap;
    word-wrap:break-word; */
    /* line-height: 28px; */
    font-size: 0.8rem;
    font-family: ${props => props.theme.fontFamily};
    color: ${p => p.theme.dark ? p.theme.primaryText : p.theme.primaryColor};
  }
`;

export const ColorPaletteWrapper = styled.div`
  position: absolute;
  bottom: -40px;
  left: -90px;

  display: flex;
  justify-content: space-around;
  background-color: ${props => props.theme.secondaryColor};
  box-shadow: ${props => props.theme.shadowSmall}; 
  padding: 8px;
  border-radius: 5px;

  @media ${props => props.theme.media.tablet} {
    bottom: -40px;
    left: 50%;
    transform: translate(-50%, 0);
  }
`;

export const ColorBox = styled.div`
  width: 40px;
  height: 40px;
  margin: 3px;
  border-radius: 5px;
`;
