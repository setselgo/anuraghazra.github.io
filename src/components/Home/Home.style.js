import styled from 'styled-components';

export const Intro = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  
  height: 80vh;
  ${props => props.theme.spacing.sectionBottom};
  
  .home__text {
    color: ${props => props.theme.primaryBlack};
    z-index: 1;

    button {
      margin: 30px 0;
    }
    
    @media ${props => props.theme.media.tablet} {
      text-align: center;
      display: flex;
      align-items: center;
      flex-direction: column;
    }
  }

  
  h1 {
    font-size: 2.5rem;
    font-weight: 900;
    text-shadow: 0 0 2px 2px white;
  }

  p {
    font-family: ${props => props.theme.fontFamily};
  }
  .adjust {
    font-size: 1.06rem;
  }
  
  .home__CTA {
    width: max-content;
  }
  .home__social {
    color: ${props => props.theme.primaryColor};
    display: flex;
    justify-content: space-between;
    font-size: 24px;
  }

  
  @media ${props => props.theme.media.tablet} {
    justify-content: space-evenly;
    flex-direction: column;
  }
`

export const HomeWrapper = styled.section`
  margin-bottom: 100px;

  .svg-rect {
    width: 40%;
    position: absolute;
    top: 50px;
    right: 0;
    z-index: -1;
  }

  @media ${props => props.theme.media.tablet} {
    .svg-rect {
      opacity: 0.8;
    }
  }
`
