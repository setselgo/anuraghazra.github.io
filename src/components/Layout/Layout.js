import React, { useState, useEffect, useRef } from "react"
import PropTypes from "prop-types"

import "normalize.css";
import { ThemeProvider } from 'styled-components';
import { themelight, themedark } from './theme';

import Wrapper from "src/components/common/Wrapper";
import Navbar from "./Navbar/Navbar";
import Footer from "./Footer";
import GlobalStyle from "src/styles/GlobalStyle";

import 'src/components/fontLib';
import useDarkMode from 'src/hooks/useDarkMode';

const Layout = ({ children }) => {
  const [theme, toggleTheme, toggleRef] = useDarkMode();

  return (
    <ThemeProvider theme={theme === 'light' ? themelight : themedark}>
      <>
        <GlobalStyle />
        <Navbar toggleRef={toggleRef} handleDarkModeToggle={toggleTheme} />

        <button onClick={toggleTheme}>toggle-theme</button>
        <Wrapper style={{ marginTop: 100, marginBottom: 50, minHeight: 'calc(100vh - 125px)' }}>{children}</Wrapper>
        <Footer />
      </>
    </ThemeProvider>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
