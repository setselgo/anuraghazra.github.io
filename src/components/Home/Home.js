import React from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import svgRect from 'src/static/home_rect.svg'

import { HeroCard } from './HeroCard';
import { HomeWrapper, Intro } from './Home.style';

import IconLink from 'src/components/common/IconLink';
import PageHeader from 'src/components/common/PageHeader';
import Flex from "src/components/common/Flex";
import Button from "src/components/common/Button";

import { Card, CardIcon, CardText, CardTitle } from "src/components/common/Card";

const ThingsILove = () => (
  <Flex justify="space-between" align="center">
    <Card>
      <CardIcon><FontAwesomeIcon icon="code" /></CardIcon>
      <CardTitle>FrontEnd</CardTitle>
      <CardText>
        i'm more front end focused and love to work with React as well as pure HTML, CSS
      </CardText>
    </Card>

    <Card>
      <CardIcon><FontAwesomeIcon icon={["fab", "js"]} /></CardIcon>
      <CardTitle>Javascirpt</CardTitle>
      <CardText>
        I just extreamly love javascirpt, i can’t even express how much i love javascirpt with just a few lines.
      </CardText>
    </Card>

    <Card>
      <CardIcon><FontAwesomeIcon icon="paint-brush" /></CardIcon>
      <CardTitle>Creative Coding</CardTitle>
      <CardText>
        I love creative coding because i do both coding & designing. making beautifull art with code is very satisfying to me
      </CardText>
    </Card>
  </Flex>
);

const Home = () => {
  return (
    <HomeWrapper id="home">
      <img className="svg-rect" src={svgRect} alt=""></img>

      <Intro>
        {/* <Parallax y={[50, -50]} className="home__text"> */}
        <div className="home__text">
          <p>Hello, i’m</p>
          <h1>ANURAG HAZRA</h1>
          <p className="adjust">CREATIVE FRONT-END WEB DEVELOPER</p>

          <div className="home__CTA">
            <Button className="cta" as="a" href="#">Download Resume</Button>

            <div className="home__social">
              <IconLink label="github" icon={["fab", "github"]} href="//github.com/anuraghazra" />
              <IconLink label="twitter" icon={["fab", "twitter"]} href="//twitter.com/anuraghazru" />
              <IconLink label="codepen" icon={["fab", "codepen"]} href="//codepen.io/anuraghazra" />
            </div>
          </div>
        </div>
        {/* </Parallax> */}
        <HeroCard />
      </Intro>

      {/* Things I LOVE */}
      <PageHeader style={{ marginBottom: 30 }}>Things i love <i className="fas fa-heart" /></PageHeader>
      <ThingsILove />

    </HomeWrapper>
  )
}

export default Home;