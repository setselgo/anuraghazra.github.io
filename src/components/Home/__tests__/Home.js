import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, cleanup } from '@testing-library/react';

import Home from '../Home';
import { themelight } from '../../Layout/theme';

afterEach(cleanup);

test('home renders correctly', () => {
  let { getByText } = render(
    <ThemeProvider theme={themelight}>
      <Home name={name} />
    </ThemeProvider>
  );

  getByText('ANURAG HAZRA');
})

