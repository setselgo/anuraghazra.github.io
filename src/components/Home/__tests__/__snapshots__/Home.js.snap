// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home renders correctly 1`] = `
<DocumentFragment>
  <section
    class="sc-EHOje jxBrYE"
    id="home"
  >
    <img
      alt=""
      class="svg-rect"
      src="test-file-stub"
    />
    <section
      class="sc-ifAKCX gJQnGm"
    >
      <div
        class="home__text"
      >
        <p>
          Hello, i’m
        </p>
        <h1>
          ANURAG HAZRA
        </h1>
        <p
          class="adjust"
        >
          CREATIVE FRONT-END WEB DEVELOPER
        </p>
        <div
          class="home__CTA"
        >
          <a
            class="sc-htoDjs iyZMqX cta"
            href="#"
          >
            Download Resume
          </a>
          <div
            class="home__social"
          >
            <a
              aria-label="github"
              href="//github.com/anuraghazra"
              target="__blank"
              title="github"
            />
            <a
              aria-label="twitter"
              href="//twitter.com/anuraghazru"
              target="__blank"
              title="twitter"
            />
            <a
              aria-label="codepen"
              href="//codepen.io/anuraghazra"
              target="__blank"
              title="codepen"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa cCATg"
      >
        <div
          class="sc-bwzfXH boMUdc"
        >
          <pre>
            1   class 
            <b>
              Person
            </b>
             {
          </pre>
          <pre>
            2         constructor() {
          </pre>
          <pre>
            3             this.name = "
            <b>
              Anurag Hazra
            </b>
            ";
          </pre>
          <pre>
            4             this.traits = ["
            <b>
              DESIGN
            </b>
            ", "
            <b>
              DEV
            </b>
            "];
          </pre>
          <pre>
            5             this.age = new Date().getFullYear() - 2001;
          </pre>
          <pre>
            6         }
          </pre>
          <pre>
            7   }
          </pre>
        </div>
        <div
          class="sc-htpNat cXzmwh"
        >
          <div
            class="sc-bxivhb frwLng"
            style="background: rgb(109, 131, 242);"
          />
          <div
            class="sc-bxivhb frwLng"
            style="background: rgb(106, 152, 240);"
          />
          <div
            class="sc-bxivhb frwLng"
          />
          <div
            class="sc-bxivhb frwLng"
            style="background: rgb(56, 56, 56);"
          />
          <div
            class="sc-bxivhb frwLng"
            style="background: rgb(199, 208, 255);"
          />
        </div>
      </div>
    </section>
    <h2
      class="sc-bZQynM gNVHZL"
      style="margin-bottom: 30px;"
    >
      Things i love 
      <i
        class="fas fa-heart"
      />
    </h2>
    <div
      class="sc-gzVnrw kAdVJI"
    >
      <div
        class="sc-iwsKbI dhJcPj"
      >
        <div
          class="sc-gZMcBi jgqArj"
        />
        <h3
          class="sc-gqjmRU jlixfV"
        >
          FrontEnd
        </h3>
        <p
          class="sc-VigVT hTTMiL"
        >
          i'm more front end focused and love to work with React as well as pure HTML, CSS
        </p>
      </div>
      <div
        class="sc-iwsKbI dhJcPj"
      >
        <div
          class="sc-gZMcBi jgqArj"
        />
        <h3
          class="sc-gqjmRU jlixfV"
        >
          Javascirpt
        </h3>
        <p
          class="sc-VigVT hTTMiL"
        >
          I just extreamly love javascirpt, i can’t even express how much i love javascirpt with just a few lines.
        </p>
      </div>
      <div
        class="sc-iwsKbI dhJcPj"
      >
        <div
          class="sc-gZMcBi jgqArj"
        />
        <h3
          class="sc-gqjmRU jlixfV"
        >
          Creative Coding
        </h3>
        <p
          class="sc-VigVT hTTMiL"
        >
          I love creative coding because i do both coding & designing. making beautifull art with code is very satisfying to me
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
